src/clockcache.c

```diff
@@ -1873,33 +1877,41 @@ clockcache_init(clockcache        *cc,   // OUT
    return STATUS_NO_MEMORY;
 }
 
+/*
+ * De-init the resources allocated to initialize a clockcache.
+ * This function may be called to deal with error situations, or a failed
+ * clockcache_init(). So check for non-NULL handles before trying to release
+ * resources.
+ */
 void
 clockcache_deinit(clockcache *cc) // IN/OUT
 {
    platform_assert(cc != NULL);
 
-   /*
-    * Check for non-null cause this is also used to clean up a failed
-    * clockcache_init
-    */
    if (cc->logfile) {
       clockcache_log(0, 0, "deinit %s\n", "");
 #if defined(CC_LOG) || defined(ADDR_TRACING)
       platform_close_log_file(cc->logfile);
 #endif
    }
 
-   platform_buffer_deinit(&cc->rc_bh);
+   if (cc->lookup) {
```

Comment:
I did not bother to arrange these backout codes in `reverse` order of how they are done in `init()` method as each backout fragment is protected under an `if()` check. 

So, although, not in "pure backout form", this ordering should work.